### ISSUE TITLE
Add OIDC authentication or error message about support authentication methods

### ISSUE THREAD

**Issue by briedel** [kubernetes-client/python#368]
I was trying to use the python client against a kubernetes cluster that uses OIDC authentication. The python-client tried to log in as an "anonymous" user, which failed. It would be good to add OIDC authentication or add an error message that the authentication method was not support.

**Comment by mbohlool** [comment]
is not already implemented in https://github.com/kubernetes-client/python-base/pull/31? try 4.0.0a client.

**Comment by mbohlool** [comment]
https://github.com/kubernetes-client/python-base/pull/31 is implementing this.

**Comment by briedel** [comment]
Thanks for the hint! I used the code from https://github.com/kubernetes-client/python-base/pull/31 and found that there is a difference between how `kubectl` behaves and the python API. When a refresh token is necessary, `kubectl` does not require `idp-certificate-authority-data` to be set:

```
    if cfg[cfgCertificateAuthorityData] != "" {
        certAuthData, err = base64.StdEncoding.DecodeString(cfg[cfgCertificateAuthorityData])
        if err != nil {
            return nil, err
        }
    }
```

The error that is called seems to be suppressed somehow. Yet, with the same config the new oidc authentication requires `idp-certificate-authority-data` to be set:

```ConfigException: Invalid kube-config file. Expected key idp-certificate-authority-data in kube-config/users[...]/user/auth-provider/config```

There is a `refresh-token` URL provided in the config though, which might be what allows `kubectl` to get around the need for `idp-certificate-authority-data`

**Comment by lichen2013** [comment]
This issue is the same as https://github.com/kubernetes-client/python/issues/195

**Comment by benley** [comment]
It appears this may be addressed via kubernetes-client/python-base#48; should this issue be closed?
